src/pages/reset.astro

```diff
@@ -0,0 +1,124 @@
+---
+export const prerender = false;
+
+import '../styles/styles.css';
+import { getAuth } from "firebase-admin/auth";
+import { sendPasswordResetEmail } from "firebase/auth";
+import { app } from "../lib/firebase_server.ts";
+import { actions } from "astro:actions";
+import Button from '../components/Button.astro';
+import BackgroundTheme from '../components/BackgroundTheme.astro';
+---
+
+<div class="settings-container">
+  <div class="settings-form">
+    <label for="email-input">Email Address:</label>
+    <input type="email" id="email-input" placeholder="Enter your email" />
+    <button class="submit-button" id="reset-password-button">Reset Password</button>
+    <p id="reset-message" class="hidden"></p>
+  </div>
+
+</div>
+
+<Button
+  id="backButton"
+  size="lg"
+  backgroundColor="#4ea7f5"
+  textColor="white"
+  borderColor="transparent"
+  href="/login"
+>
+  Back
+</Button>
+
+<style>
+    .settings-container {
+      padding: 2rem;
+      max-width: 600px;
+      margin: auto;
+      display: flex;
+      flex-direction: column;
+      gap: 2rem;
+    }
+  
+    .settings-form {
+      display: flex;
+      flex-direction: column;
+      gap: 1rem;
+    }
+
+    #email-input {
+      width: 100%;
+      padding: 1rem;
+      font-size: 1.2rem;
+    }
+  
+    .submit-button {
+      background-color: skyblue;
+      color: white;
+      padding: 1rem;
+      border: none;
+      cursor: pointer;
+      border-radius: 5px;
+      font-size: 1.2rem;
+    }
+  
+    .submit-button:hover {
+      background-color: #1d99db;
+    }
+  
+    .hidden {
+      display: none;
+    }
+    .success {
+      color: rgb(107, 172, 230);
+    }
+  
+    .error {
+      color: black;
+    }
+  </style>
+<script>
+  
+import '../styles/styles.css';
+import { getAuth, sendPasswordResetEmail } from "firebase/auth";
+import { app } from "../lib/firebase_client.ts";
+import { actions } from "astro:actions";
+
+  const auth = getAuth(app); 
+
+  const resetButton = document.getElementById("reset-password-button");
```

Comment:
You should `document.getElementById("reset-password-button")!` to ditch the `if`s later, as the element is guaranteed to exist by this point